Adding docs for the Localized dataset on the QFieldCloud (#557)
opengisch/QField-docs: documentation/how-to/outside-layers.en.md

tx_slug: documentation_how-to_outside-layers
---

# Shared datasets

It is possible to use layers stored in a single location—referred to as a localized data path—across multiple projects. This can help reduce storage requirements for large datasets such as orthophoto raster files, land use vector files, etc., as well as ease the management of dataset updates.

> There are two main approaches to using shared local datasets: a manual copying method for into devices running QField, and a more integrated approach using QFieldCloud.

Author: Many thanks for the suggestions~

# Shared local datasets

> It is possible to use a layer which is stored outside the project folder. This is useful if a basemap (e.g., a large orthophoto) or large, unmodified vector data is used in more than one project. By sharing datasets, features and their attributes remain identifiable across projects.

Reviewer: I think it'd be good here to refer to the primary reason why shared datasets are useful, namely the de-duplication of large files being stored.

```suggestion
It is possible to use layers stored in a single location - referred to as a localized data path - across multiple projects. This can help reduce storage requirement for large datasets such as orthophoto raster files, land use vector files, etc. as well as easing management of dataset updates.
```

Author: Yup, great idea!

> There are two main approaches to using shared local datasets: a manual copying method for direct use in QField, and a more integrated approach using QFieldCloud which leverages QGIS's Localized Data Paths.

Reviewer: Both methods leverages QGIS' localized data paths :)

```suggestion
There are two main approaches to using shared local datasets: a manual copying method into devices running QField, and a more integrated approach using QFieldCloud.
```

> ## Manual Copying of Shared Datasets for QField

Reviewer: ```suggestion
## Managing Localized Data Paths in QGIS
```

> This method is suitable for users who want to manually manage shared files on their devices, often for basemaps, without using QFieldCloud for these specific files.

Reviewer: ```suggestion
When building projects for QField, make sure the datasets you want to share across multiple projects are within localized data paths in QGIS. 
```

:material-monitor: Desktop preparation

> In QGIS, it's good practice set where your shared datasets are located.
> 1. Open QGIS.
> 2. Go to *Options > Data Sources > Localized Data Paths*.
> 3. Add the path(s) to your external data.
> This primarily helps QGIS itself locate these layers (e.g., if you move projects between computers or use relative paths) and is **essential for the QFieldCloud workflow described later.**
> 
> !![QGIS Options Dialog showing Data Sources and Localized Data Paths setup](../assets/images/external_path.png)

Reviewer: ```suggestion

In QGIS, go to the Options dialog's data sources panel, where you find a localized data paths section. From there, you can add a number of path(s) within which datasets will be treated as shared datasets. This must be done as a preliminary step to building your projects, including those to be uploaded to QFieldCloud.

!![QGIS Options Dialog showing Data Sources and Localized Data Paths setup](../assets/images/external_path.png)
```

### Using Manually Copied Datasets in QField

> To use these shared datasets manually in QField (e.g., as basemaps without using QFieldCloud for them):

Reviewer: Manually copying files can also be used by cloud projects, we shouldn't imply otherwise.

```suggestion
To use these shared datasets manually in QField, locate the QField app directories on your device:
```

> 1.  **Locate the QField app directories on your device:**
>     - In QField, open any project.
>     - Open the Dashboard (hamburger icon), and open the main menu (three dots).
>     - Tap 'About QField'.
>     - The application directory locations will be at the bottom as 'App directories' path (paths differ by operating system).
> 2.  **Copy your shared dataset files** (e.g., the orthophoto GeoTIFF or vector GeoPackage) into the `(Your App Directory)/QField/basemaps` subfolder on your device. QField automatically scans this folder for basemaps and other recognizable data.

Reviewer: ```suggestion
- In QField, open any project.
- Reveal the side dashboard by clicking on the hamburger icon, and open the main menu by clicking on the 3-dot button.
- Select the 'About QField' menu item.
- The application directory locations will be at the bottom just below 'App directories' path (paths differ by operating system).

Once you have located the directory, copy your shared dataset files into the `[App Directory]/QField/basemaps` folder on your device. QField will automatically scans this folder for basemaps and other recognizable data.
```

Author: Great!... Much better!

> To use layers from localized data paths on your device, add the datasets to the folder `<drive>:/Android/data/ch.opengis.qfield/files/QField/basemaps`.

Reviewer 2: This is Android specific
Could we rethink how we document this?
I think we want one generic documentation of where to find the qfield data path on your operating system. And in here only keep the relative path from there.

Reviewer 2: We also want to make it clear that there are two options

1. manual copying
2. qfieldcloud based approach

At the moment it's not clear, that this information is irrelevant if QFieldCloud is used.

Author: Addressed!.... Many thanks!

!!! note

>     Since QField 2, the basemaps files needs to be stored in the app directory `<drive>:/Android/data/ch.opengis.qfield/files/QField/basemaps` instead of the devices main directory `<drive>:/QField/basemaps`.

Reviewer 2: I think we want to get rid of this legacy information altogether.

Author: I agree, the Note should be completely removed!

Author: Great suggestion!

## Managing Shared Datasets with QFieldCloud

> QFieldCloud significantly enhances the management of shared localized datasets. This allows you to upload these common datasets once and reuse them across multiple cloud projects:

Reviewer 2: We should focus this page on this method. I.e. this should be the "default" method (not the "enhanced" method)

!!! note

>     Since QField 2 the basemaps files needs to be stored in the app directory `<drive>:/Android/data/ch.opengis.qfield/files/QField/basemaps` instead of the devices main directory `<drive>:/QField/basemaps`.

Reviewer 3: ```suggestion
    Since QField 2, the basemaps files needs to be stored in the app directory `<drive>:/Android/data/ch.opengis.qfield/files/QField/basemaps` instead of the devices main directory `<drive>:/QField/basemaps`.
```

## Managing Shared Datasets with QFieldCloud

> QFieldCloud significantly enhances the management of shared localized datasets. This allows you to upload these common datasets once and use them on multiple cloud projects:

Reviewer 3: ```suggestion
QFieldCloud significantly enhances the management of shared localized datasets. This allows you to upload these common datasets once and reuse them across multiple cloud projects:
```

### 1. Prepare Your QGIS Project with Localized Datasets

**Follow the same [Desktop Preparation (QGIS)](#add-a-layer-outside-of-exported-project-folder)**:

- Critically, make sure the Localized Data Paths in QGIS Options are correctly configured to point to the location of these shared datasets on your computer. This tells QFieldSync which files to treat as "localized" for cloud handling.
- Ensure your localized layers are part of your QGIS project.

### 2. Upload Localized Datasets to QFieldCloud

> **Create/Configure Your Cloud Project**:

Reviewer 3: ```suggestion
**Create or Configure Your Cloud Project**:
```
It's slighty more formal this way

## 3. Localized datasets within QFieldCloud web interface

Once uploaded, these localized datasets will appear in two key places on the QFieldCloud web interface:

> * Within the dedicated **`shared_datasets`** project itself.

Reviewer 4: Just noticed sometimes it is bolded and sometimes not. Just to ensure there is some kind of logic there, if there is, please ignore my comment.

$ export QFIELDCLOUD_TOKEN="TOKEN_HERE"
```

- [List the QFieldCloud projects](https://opengisch.github.io/qfieldcloud-sdk-python/examples/#list-your-projects) and get the project ID of the **`shared_datasets`** project:

```console
$ qfieldcloud-cli list-projects
Listing projects…
Projects the current user has access to:
| ID                                   | OWNER/NAME           | IS PUBLIC | DESCRIPTION     |
--------------------------------------------------------------------------------------------

> | 90e83606-dce8-4b0d-854a-388904d8a739 | USER/shared_datasets | 0         | Shared datasets |

Author: Great!.... Removed!
